### ISSUE TITLE
Sync deletes entries in Google and at the next sync the corresponding entries in Outlook

### ISSUE THREAD

**Issue by NochN-Konto** [phw198/OutlookGoogleCalendarSync#259]
Well, first big thank you for this program. Wonderful!!! It is working for me almost a year now.

Got one issue: I am used to create most of my appointments by importing them to outlook from an excel table, ca 40 at a time. This was never an issue, but now with Version 2.5.0. and I think 2.4..0 also, stayed with 2.3. for a long time..) Those imported appointments are first correctly exported to Google. Then a few syncs later (couldn't find the  right number or reason yet) all formerly synced and imported appointments are to be deleted in google. Once they are deleted, the next time you sync they will also be deleted in Outlook.

What did happen?

Do the appointments imported from excel miss some info needed?

[OGcalsync - Kopie.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/1122118/OGcalsync.-.Kopie.txt)

**Comment by Lunarmanpl** [comment]
Hi
I have similar problem using Your program. I am using it to sync outlook calendar between two PC and Android phone via google calendar. Problem is that if I add something on second PC while first one is offline then after sync with first one it want to delete that added in google. Its usually ends with empty calendar in google and outlook. I don't notice that problem when adding to google anything via phone, its look like it has some problem with notifying in google with appointments were created/modified or maybe reading which one were modified since last sync.
Program often in first sync after turning on PC is checking which appointments in google are not in outlook and want to delete them. Next look which are not in outlook that are in google and delete those too. For now i had to disable deleting in program and its sync without problem but i have to delete it manualy.

[OGcalsync.log.2017-07-13.txt](https://github.com/phw198/OutlookGoogleCalendarSync/files/1164621/OGcalsync.log.2017-07-13.txt)

**Comment by phw198** [comment]
@NochN-Konto 
It seems like things have got a bit confused between upgrading and downgrading between v2.3, v2.4 and v2.5. To clean things up:
1. From v2.5, go to _About_ and click _Check for Alphas_ and upgrade to v2.5.2
1. Change the sync from Outlook to Google
1.1. Choose _Merge_ and _Confirm Deletes_
1. Shift-click the _Sync_ button to force a compare of every item and sync into Google
1. Return to a two-way sync

Let me know if this resolve further problems.

**Comment by phw198** [comment]
@Lunarmanpl 
OGCS isn't designed to work with having a central Google calendar syncing to two separate Outlook calendars on different computers, most likely with two-way sync for each (I think this is what you're describing). 

Calendar items in Google will have attributes associated with them that links them to the Outlook calendar that first synced them. By then adding a second calendar to the mix, it will fight over which is the "master" Outlook calendar that is syncing the Google items - and you'll likely get odd things happens such as you describe. Sorry!

**Comment by NochN-Konto** [comment]
Paul, this is amazing. I'm getting personal support, I can't believe
this. 😎 Thank you so much. It works fine up to now...

This is worth a donation, these days I will send you something...

God Bless You...


2017-07-23 12:39 GMT+02:00 Paul <notifications@github.com>:

**Comment by phw198** [comment]
@NochN-Konto :slightly_smiling_face: Do I take it that the "cleanup" steps helped?

**Comment by NochN-Konto** [comment]
Hi Paul,
unfortunately it did not help for long. I think I know what the problem is.
The guy who wrote about his two different computers, that he wants to sync
with only one google account, made me think.

I use a remote desktop connection to the central servers of my company.
Since we have many users, there are a couple of physically separated
servers, which are chosen by an algorithm to distribute users equally.  So,
when I open a connection I get one physical computer and the next time I
open a new session I get another physical computer, I can't interfere with
that.

Looks like with version 2.3. or so of OGCS the way google appointments are
identified is somewhat "towed" to hardware uniqueness. In fact I do think I
have the same problem as lunarmanpl and his two computers. With the
exception I do have about ten computers or so...

I think the best solution is to delete by hand. But I think that way any
changes in outlook won't make it to the google calendar anymore, exept i am
in the same remote session. Well or, once a new session starts I do all the
time the shift click and let everything be deleted and resynced. But then I
miss the changes I made on my mobile phone... or i use an older version. Do
you know with what version the hardware based identifier started?

Or what. Please enlighten me ☺

thanks for all.

2017-07-30 23:31 GMT+02:00 Paul <notifications@github.com>:

**Comment by phw198** [comment]
@NochN-Konto No, there's no nothing in OGCS that would tie the calendar entries to a specific machine, so I'm not sure this is quite the problem. However, explain to me a bit more how this remote desktop / virtualisation works. Do you know if it's something like a Citrix remote desktop cluster, or your more classic VM virtualisation with your Windows session spun up on demand on one of the load balanced hosts? 

Either way, does your Windows session follow you around - eg if you haven't logged out and your session drops, the next time you connect, is your session still alive but maybe on a different host server? Just wondering if it's possible you have multiple Windows sessions at the same time, possibly with contending OGCS applications also running.

Basically, when you sync from Outlook to Google, the object ID* of the Outlook appointment gets written to the Google calendar item, Then on subsequent syncs, if there is no longer an appointment in Outlook with that ID, the Google item will be deleted. What you could do is turn on CVS exports and then after an import to Outlook see what the object IDs are - make a copy of the CSVs. Then when unexpected deletes happen in Google, see what the entries/object IDs in the CSVs are and if they've changed.

\* There are actually 3 Outlook IDs: Object ID, GlobalAppointmentID and CalendarID. I actually suspect one of these IDs is changing due to your virtualisation configuration.

**Comment by NochN-Konto** [comment]
Hi Paul,

well thank you for your answer.

Yes the remote Desktop is established via a citrix remote application. I do
have a login account, which I can use from any computer in the corporate
Intranet. It is usually my business laptop, but it I can login from any
other computer in the intranet, having the citrix application installed.
Yes, session follows me when moving onto another computer (client in that
case I would say).

Starting a complete new session in the morning leads me to a server (host I
would say) machine out of many, the one which is less busy at the moment.

Synching in a established session always works. No matter how often and
where I do it. Initiating a new session brings up the problem.

In addition, we use a zarafa mail server hooked to outlook, but they are
working on a change to MS Exchange.

I will do the cvs export to check when the ID changes. It must happen at
some point...

Thanks for help. I like it ☺

2017-08-06 12:28 GMT+02:00 Paul <notifications@github.com>:
